oslo_messaging/metrics.py

```diff
@@ -0,0 +1,22 @@
+import logging
+import threading, queue
+import statsd
+
+LOG = logging.getLogger(__name__)
+
+class Metrics():
+   def __init__(self, host, port):
+      self._port = port
+      self.queue = queue.Queue()
+      self.statsd = statsd.StatsClient(host, port)
+
+   def start(self):
+      threading.Thread(target=self.counter, daemon=True).start()
+
+   def counter(self):
+      while True:
+         event = self.queue.get()
+         method = event.message.get('method')
+
+         stat = "oslo.messaging.{}.{}".format(method, event.message.tag)
```

Comment:
I'm a little confused how that works. I thought we put a dictionary into the queue - which would end up as `event` here. A dictionary doesn't support `event.message` though, but only supports `event['message']` - at least in Python 2, which rocky is still using.

Additionally, I would have assumed that we want `event['tag']` here instead of `event['message']['tag']`.

Reply:
hm yes me too. i cherry picked the wrong commit. it should be like this:
```
  incoming = self.queue.get()
  message = incoming['message'].message

  method = message.get('method')
  stat = "oslo.messaging.{}.{}".format(method, incoming['tag'])
  self.statsd.incr(stat)
  self.queue.task_done()
```

Reply:
```python
  self.statsd.incr(stat)
  self.queue.task_done()
```
This being at the end, if there's ever a message that's misformatted, we would run into a problem again and again for the same message, because it stays in the queue, right?

Additionally, if there's ever a problem with a message or reaching statsd, our thread will just die. Might make sense to add a `try-except` around the message-handling that imho. We could then think about whether we want to ignore a mis-formatted message and just ack anything in `finally` and lose that info or what we want to do in that case.